### PR TITLE
Prepping flight_plan module to have non-repudiation

### PR DESCRIPTION
**Pull request by cyapa**
## Proposed changes

Describe the big picture of the code changes here to the team. If it fixes a bug or resolves a feature request, be sure to add the JIRA link to that issue.

## Types of changes
[JIRA](https://ssrc.atlassian.net/browse/SCUTM-292)

Extracted the logic inside the big function `set_flight_declaration` so that we can reuse it when we implement `signed_set_flight_declaration`: Supports non-repudiation.

Previously the code logics of:

-  _parsing JSON payload_, 
- _getting Operational Intent_ 
- _sending RABBITMQ messages_ were bundled together.

Extracted 3 private methods from above code so that we can reuse the same code when implementing `POST signed_set_flight_declaration`

Extracted 3 functions are:

- _parse_flight_declaration_request()
- _get_operational_intent()
- _send_fd_creation_notifications()

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply._

- [x] Lint and unit tests pass locally with the changes
- [x] Added new tests that prove the changes in this PR works
- [ ] Added necessary documentation (if appropriate)
- [ ] Added copyrights to new files (if appropriate)

## Further comments

_Add special comments about the changes here if required._

### CODE REVIEW

File: flight_declaration_operations/serializers.py [hunk 1]
```diff
@@ -39,8 +39,8 @@ class FlightDeclarationRequestSerializer(serializers.Serializer):
     originating_party = serializers.CharField(
         required=False, default="No Flight Information"
     )
-    start_datetime = serializers.DateTimeField()
-    end_datetime = serializers.DateTimeField()
+    start_datetime = serializers.DateTimeField(required=False, default=None)
```
Comment (cyapa) [inline flight_declaration_operations/serializers.py:42]: Sending flight_time is optional

File: flight_feed_operations/pki_helper.py [hunk 2]
```diff
@@ -1,120 +1,144 @@
-from .models import SignedTelmetryPublicKey
```
Comment (cyapa) [inline flight_feed_operations/pki_helper.py:1]: In this file, the changes are only formatting the code lines. :)

File: flight_declaration_operations/views.py [hunk 3]
```diff
@@ -267,6 +293,16 @@ def set_flight_declaration(request: HttpRequest):
     return HttpResponse(op, status=status.HTTP_200_OK, content_type="application/json")
 
 
+@api_view(["POST"])
+@requires_scopes(["blender.write"])
+def set_signed_flight_declaration(request: HttpRequest):
+    return HttpResponse(
+        json.dumps({"status": "_TO_BE_IMPLEMENTED_"}),
```
Comment (govindsi) [inline flight_declaration_operations/views.py:300]: Looks good to me with assumption that implementation will come in next PR.

**Review by govindsi — APPROVED** [review]

File: flight_declaration_operations/serializers.py (hunk 1 above)
Comment (lfonosol) [reply flight_declaration_operations/serializers.py:42]: Do we need to have an optional end_datetime as well?

Comment (cyapa) [reply flight_declaration_operations/serializers.py:42]: `end_datetime` is also optional :)

Comment (lfonosol) [reply flight_declaration_operations/serializers.py:42]: Oh yeah there it is :-D

**Pull request merged**